lib/src/supabase_auth.dart

```diff
@@ -62,40 +79,103 @@ class SupabaseAuth {
     LocalStorage localStorage = const HiveLocalStorage(),
     String? authCallbackUrlHostname,
   }) async {
-    _instance._initialized = true;
-    _instance._localStorage = localStorage;
-    _instance._authCallbackUrlHostname = authCallbackUrlHostname;
-
-    _instance._authSubscription =
-        Supabase.instance.client.auth.onAuthStateChange((event, session) {
-      _instance._onAuthStateChange(event, session);
-      if (!_instance._listenerController.isClosed) {
-        _instance._listenerController.add(event);
-      }
-    });
+    try {
+      _instance._initialized = true;
+      _instance._localStorage = localStorage;
+      _instance._authCallbackUrlHostname = authCallbackUrlHostname;
+
+      _instance._authSubscription =
+          Supabase.instance.client.auth.onAuthStateChange((event, session) {
+        _instance._onAuthStateChange(event, session);
+        if (!_instance._listenerController.isClosed) {
+          _instance._listenerController.add(event);
+        }
+      });
 
-    await _instance._localStorage.initialize();
+      await _instance._localStorage.initialize();
 
-    final hasPersistedSession = await _instance._localStorage.hasAccessToken();
-    if (hasPersistedSession) {
-      final persistedSession = await _instance._localStorage.accessToken();
-      if (persistedSession != null) {
-        final response = await Supabase.instance.client.auth
-            .recoverSession(persistedSession);
+      final hasPersistedSession =
+          await _instance._localStorage.hasAccessToken();
+      if (hasPersistedSession) {
+        final persistedSession = await _instance._localStorage.accessToken();
+        if (persistedSession != null) {
+          final response = await Supabase.instance.client.auth
+              .recoverSession(persistedSession);
+          final error = response.error;
 
-        if (response.error != null) {
-          Supabase.instance.log(response.error!.message);
+          if (error != null) {
+            Supabase.instance.log(response.error!.message);
+            if (!_instance._initialSessionCompleter.isCompleted) {
+              _instance._initialSessionCompleter.completeError(error);
+            }
+          }
+          if (!_instance._initialSessionCompleter.isCompleted) {
+            _instance._initialSessionCompleter.complete(response.data);
+          }
         }
       }
-    }
+      WidgetsBinding.instance?.addObserver(_instance);
+      _instance._startDeeplinkObserver();
 
-    return _instance;
+      if (!_instance._initialSessionCompleter.isCompleted) {
+        // Complete with null if the user did not have persisted session
+        _instance._initialSessionCompleter.complete(null);
+      }
+      return _instance;
+    } catch (e) {
+      if (!_instance._initialSessionCompleter.isCompleted) {
+        _instance._initialSessionCompleter.completeError(e);
+      }
+      rethrow;
```

Comment:
It should complete with the stacktrace

```suggestion
    } catch (error, stacktrace) {
      if (!_instance._initialSessionCompleter.isCompleted) {
        _instance._initialSessionCompleter.completeError(error, stracktrace);
      }
      rethrow;
```